src/robot/libraries/String.py

```diff
@@ -121,6 +122,47 @@ def decode_bytes_to_string(self, bytes, encoding, errors='strict'):
             raise TypeError('Can not decode strings on Python 3.')
         return bytes.decode(encoding, errors)
 
+    def format_string(self, template_or_str, *posicional_search_replace,
+                      **named_search_replace):
+        """Formats a ``string`` using the given ``posicional_search_replace``
+        and ``named_search_replace``.
+
+        If the given ``string`` is a valid file path (absolute or relative), opens the file in read
+        mode and then format its content using the given
+        ``posicional_search_replace`` and ``named_search_replace``.
+
+        This keyword uses python's string format. For more information see:
+        [https://docs.python.org/2.7/library/string.html#formatstrings]|Python2 format syntax]
+        or
+        [https://docs.python.org/3.6/library/string.html#formatstrings]|Python3 format syntax]
+
```

Comment:
There's a very good 3rd party formatting resource, which you might consider adding here - https://pyformat.info.  
It's quite in-depth though, I'm not sure won't it be a bit too intimidating for RF users w/o python background.